src/NServiceBus.Core.Tests/AssemblyScanner/AssemblyScannerTests.cs

```diff
@@ -173,50 +192,63 @@ public void AppDomainAssemblies_are_included_when_enabling_ScanAppDomainAssembli
 
             var result = scanner.GetScannableAssemblies();
 
-            Assert.IsTrue(result.Assemblies.Contains(busAssembly));
+            Assert.IsTrue(result.Assemblies.Contains(appDomainAssembly));
+        }
+
+        [Test]
+        [RunInApplicationDomain]
+        public void FileSystemAssemblies_are_excluded_when_disabling_ScanFileSystemAssemblies()
+        {
+            var busAssembly = new DynamicAssembly("Fake.NServiceBus.Core");
+
+            var scanner = new AssemblyScanner(DynamicAssembly.TestAssemblyDirectory)
+            {
+                CoreAssemblyName = busAssembly.DynamicName,
+                ScanAppDomainAssemblies = false,
+                ScanFileSystemAssemblies = false
+            };
+
+            var result = scanner.GetScannableAssemblies();
+
+            Assert.IsFalse(result.Assemblies.Contains(busAssembly));
         }
 
         [Test]
         [RunInApplicationDomain]
         public void Does_not_throw_exception_when_scanning_duplicate_assemblies()
         {
             var busAssembly = new DynamicAssembly("Fake.NServiceBus.Core");
+            var duplicateAssembly = new DynamicAssembly("DuplicateAssembly", references: new[] { busAssembly });
 
             Directory.CreateDirectory(Path.Combine(DynamicAssembly.TestAssemblyDirectory, "subdir"));
-            var destFileName = Path.Combine(DynamicAssembly.TestAssemblyDirectory, "subdir", busAssembly.FileName);
+            var destFileName = Path.Combine(DynamicAssembly.TestAssemblyDirectory, "subdir", duplicateAssembly.FileName);
             // create a duplicate of the scanned assembly in a subfolder:
-            File.Copy(busAssembly.FilePath, destFileName);
+            File.Copy(duplicateAssembly.FilePath, destFileName);
 
             var scanner = new AssemblyScanner(DynamicAssembly.TestAssemblyDirectory)
             {
                 ScanNestedDirectories = true,
                 CoreAssemblyName = busAssembly.DynamicName
             };
 
-            Assert.DoesNotThrow(() => scanner.GetScannableAssemblies());
+            var result = scanner.GetScannableAssemblies(); // should not throw
+            Assert.IsTrue(result.Assemblies.Contains(duplicateAssembly));
         }
 
+        [Ignore("can't force an actual error")]
```

Comment:
this test is already not doing what it should. The current test does not cause any loading failures and only tests that the core assembly is loaded (which it no longer is by default). I haven't found a way to force an assembly loading error because it seems the runtimes now all handle most of the weird cases without exceptions. Looking for suggestions.

Reply:
Could this be related to the weird NUnit assembly resolve thing you discovered recently? If yes can it be disabled?

Reply:
I didn't see an option to disable it. I think there is a good chance it's related, but generally the .NET Core assembly resolving has been made more resilient and cases where it previously threw exceptions are now also just returning null, so it might also be SDK related.